controllers/pulp/controller.go

```diff
@@ -123,8 +123,8 @@ func (r *PulpReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 			Message:            pulp.Name + " operator tasks running",
 		})
 	}
-
-	if strings.ToLower(pulp.Spec.IngressType) != "route" && pulp.Spec.ImageVersion != pulp.Spec.ImageWebVersion {
+	needsPulpWeb := strings.ToLower(pulp.Spec.IngressType) != "route" && strings.ToLower(pulp.Spec.IngressClass) != "nginx"
```

Comment:
hum... what about clusters with ingress but without ingress class? 
or  what if users defined an ingressclass called _nginx-prod_ or _nginx-dev_ instead of _nginx_? (an example of this scenario is to reproduce the behavior we do with router sharding https://docs.openshift.com/container-platform/4.11/networking/routes/route-configuration.html#nw-ingress-sharding-route-configuration_route-configuration)

https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class
_"There are some ingress controllers, that work without the definition of a default IngressClass. For example, the Ingress-NGINX controller can be configured with a flag --watch-ingress-without-class."_

Reply:
maybe it is matter of finding a better name for the variable, I just want a variable for the user to confirm they have the nginx controller, maybe something like: `using_nginx_ingress: yes`